x-pack/plugins/security_solution/common/siem_migrations/rules/resources/splunk_identifier.test.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { splResourceIdentifier } from './splunk_identifier';
+
+describe('splResourceIdentifier', () => {
+  it('should extract macros correctly', () => {
+    const query =
+      '`macro_one` some search command `macro_two(arg1, arg2)` another command `macro_three`';
+
+    const result = splResourceIdentifier(query);
+
+    expect(result.macro).toEqual(['macro_one', 'macro_two', 'macro_three']);
+    expect(result.list).toEqual([]);
+  });
+
+  it('should extract lookup tables correctly', () => {
+    const query =
+      'search ... | lookup my_lookup_table field AS alias OUTPUT new_field | inputlookup other_lookup_list | lookup third_lookup';
+
+    const result = splResourceIdentifier(query);
+
+    expect(result.macro).toEqual([]);
+    expect(result.list).toEqual(['my_lookup_table', 'other_lookup_list', 'third_lookup']);
+  });
+
+  it('should extract both macros and lookup tables correctly', () => {
+    const query =
+      '`macro_one` some search command | lookup my_lookup_table field AS alias OUTPUT new_field | inputlookup other_lookup_list | lookup third_lookup';
+
+    const result = splResourceIdentifier(query);
+
+    expect(result.macro).toEqual(['macro_one']);
+    expect(result.list).toEqual(['my_lookup_table', 'other_lookup_list', 'third_lookup']);
+  });
+
+  it('should return empty arrays if no macros or lookup tables are found', () => {
```

Comment:
Can you add a test to make sure macros aren't accidentally found when the backtick(`) character is an escpaed part of the search string?

Reply:
That's a good point, this made me investigate possible mismatches of macros and lookup lists in SPL queries, there are two places where they are not evaluated:

- String literals: we need to ignore macro and lookup list syntax in them
- Comment blocks: they are defined with triple backticks,  \`\`\`this is a comment\`\`\`, so we need to account for them and also ignore macro and lookup list syntax inside.

I am preparing the changes to address those scenarios.